Enhanced functionality for RotationMatrix class

mitiguy opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7801)
<!-- Reviewable:end -->

amcastro-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>. Just some minor comments.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[math/rotation_matrix.h, line 190 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EXaIf7PydATo6Rj0f:-L3EXaIf7PydATo6Rj0g:b-xoicd8) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/math/rotation_matrix.h#L190)):*
> ```objc
> 
> /// Computes the rotation matrix for rotating by theta (radians) around the
> /// positive X axis.
> ```

regarding the documentation of these methods. If they return the matrix for `R_AB`, does that mean that B is rotated `theta` radians about A's X axis? or viceversa? probably worth clarifying?

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 55 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EY2uVaaxW8R87DOEU:-L3EY2uVaaxW8R87DOEV:bpvzvs1) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L55)):*
> ```objc
> 
> 
>   /// Makes the %RotationMatrix R_AB associated with a simple rotation by an
> ```

`R_AB` with single quotes?

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 60 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EYWFI3vu9d1OwVQpH:-L3EYWFI3vu9d1OwVQpI:b-c9ntr6) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L60)):*
> ```objc
>   /// @note R_AB relates two right-handed orthogonal bases A and B having unit
>   /// vectors Ax, Ay, Az and Bx, By, Bz.  Initially, Bx = Ax, By = Ay, Bz = Az,
>   /// then B is subjected to a right-handed rotation relative to A of theta*Bx.
> ```

I don't really know what theta*Bx is. Maybe say " ... of `theta` about `Bx`, where theta is defined positive in accordance to the right hand rule." (please don't forget single quotes around math symbols)

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 70 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EwKEQkvVoWEjtz0Fe:-L3EwKEQkvVoWEjtz0Ff:bc85y3w) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L70)):*
> ```objc
>   }
> 
>   /// Makes the %RotationMatrix R_AB associated with a simple rotation by an
> ```

ditto comments for `MakeRotationMatrixX()`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 78 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EvlN0qyJQdZ0sPBZw:-L3EvlN0qyJQdZ0sPBZx:b-rfrftz) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L78)):*
> ```objc
>   ///        ⎣ -sin(theta)   0   cos(theta) ⎦
>   /// ```
>   /// @see MakeRotationMatrixX
> ```

I would say: "See documentation for MakeRotationMatrixX() for definitions on the frames A and B related to this rotation and for `theta` sign convention". Otherwise I would just prefer to duplicate the discussion on A and B here to make all docs look alike since it really doesn't hurt, is not such a lengthy piece of doc anyways.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EwMnBydjDAcJVwLNV:-L3EwMnBydjDAcJVwLNW:bc85y3w) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L83)):*
> ```objc
>   }
> 
>   /// Makes the %RotationMatrix R_AB associated with a simple rotation by an
> ```

ditto comments for `MakeRotationMatrixX()`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 97 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Exdo_IFCinReMYSyI:-L3Exdo_IFCinReMYSyJ:bipx87e) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L97)):*
> ```objc
> 
>   /// Makes the %RotationMatrix for a Space-fixed (extrinsic) X-Y-Z rotation by
>   /// "roll-pitch-yaw" angles [q(0), q(1), q(2)], which is equivalent to a
> ```

math between single quotes.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 107 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey0gLVlpJUzm84MTh:-L3Ey0gLVlpJUzm84MTi:b-d15h5e) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L107)):*
> ```objc
>   /// Makes the %RotationMatrix for a Body-fixed (intrinsic) Z-Y-X rotation by
>   /// "yaw-pitch-roll" angles [q(0), q(1), q(2)], which is equivalent to a
>   /// Space-fixed X-Y-Z rotation by "roll-pitch-yaw" angles [q(2), q(1), q(0)].
> ```

singles quotes for math

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 109 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey-8ppudthAZ3wnSO:-L3Ey-8ppudthAZ3wnSP:b-d15h5e) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L109)):*
> ```objc
>   /// Space-fixed X-Y-Z rotation by "roll-pitch-yaw" angles [q(2), q(1), q(0)].
>   /// @param[in] q radian measures of three angles, namely yaw, pitch, and roll.
>   /// @note Denoting yaw y = q(0), pitch p = q(1), roll r = q(2), this method
> ```

singles quotes for math

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey23-CPTxHn0aTf1Z:-L3Ey23-CPTxHn0aTf1_:b-d7hlha) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
> `R_AB`
singles quotes for math

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EyE1QefOZOoSFDaAF:-L3EyE1QefOZOoSFDaAG:b-hneewx) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
> ```objc
>   /// This rotation sequence relates frames A and B as follows: Initially,
>   /// Bx = Ax, By = Ay, Bz = Az, then B is subjected to sucessive right-handed
>   /// rotations relative to A, characterized by y*Bz, then p*By, then r*Bx.
> ```

 "sucessive right-handed rotations relative to A" 
This is confusing. Each rotation is relative to an intermediate frame, there are four frames here:
`R_AB = R_AB1 * R_B1B2 * R_B2B`.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EyvtBr6uE-jC3Ukxs:-L3EyvtBr6uE-jC3Ukxt:b78p0wm) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
> ```objc
>   /// This rotation sequence relates frames A and B as follows: Initially,
>   /// Bx = Ax, By = Ay, Bz = Az, then B is subjected to sucessive right-handed
>   /// rotations relative to A, characterized by y*Bz, then p*By, then r*Bx.
> ```

"y*Bz, then p*By, then r*Bx." I find this notation strange. I think it actually confuses rather than clarifying anything. It might be true for differential rotations (i.e. true also for angular velocities) but not for finite rotations.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 280 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzRRn5QCsHWrbUT9V:-L3EzRRn5QCsHWrbUT9W:b2tpczo) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L280)):*
> ```objc
>   /// to check if they are exactly the same.
>   /// @param[in] other %RotationMatrix to compare to `this`.
>   /// @returns `true` if `‖this - other‖∞ = 0`.
> ```

I'd say: "@returns `true` if each element of `this` exactly equals the elements in `other`".
Mathematically both are of course equal. The difference is when we perform floating point checks

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 282 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzJ1j-Fk38haa8TBd:-L3EzJ1j-Fk38haa8TBe:b-mtp3la) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L282)):*
> ```objc
>   /// @returns `true` if `‖this - other‖∞ = 0`.
>   bool IsExactlyEqualTo(const RotationMatrix<T>& other) const {
>     return IsNearlyEqualTo(matrix(), other.matrix(), 0.0);
> ```

I think you'd like to do ` return  matrix() == other.matrix()` here?

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 124 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzsxzNVih5vnAKu35:-L3EzsxzNVih5vnAKu36:byzyr6d) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L124)):*
> ```cpp
>   m <<  1,         0,            0,
>         0, cos_theta,   -sin_theta,
>         0, sin_theta,    cos_theta;
> ```

this test could have the problem that you could've exactly copied the doc from `MakeRotationMatrixX()`.
I think a better test would compare against `Eigen::AngleAxisd(theta, Vector3d::UnitX()).matrix()`

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 141 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-RG7tV_CdHYDXYb_:-L3F-RG7tV_CdHYDXYba:b-eg4gtw) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L141)):*
> ```cpp
>        -sin_theta,    0,   cos_theta;
> 
>   RotationMatrix<double> R = RotationMatrix<double>::MakeRotationMatrixY(theta);
> ```

ditto. Test against `Eigen::AngleAxisd(theta, Vector3d::UnitY()).matrix()`?

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 156 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-UBeQnS1VJ6DDWYy:-L3F-UBeQnS1VJ6DDWYz:bi92j73) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L156)):*
> ```cpp
>                0,           0,    1;
> 
>   RotationMatrix<double> R = RotationMatrix<double>::MakeRotationMatrixZ(theta);
> ```

ditto, test against `Eigen::AngleAxisd(theta, Vector3d::UnitZ()).matrix()`?

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 169 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-_9jwxfxkrNyHixJ:-L3F-_9jwxfxkrNyHixK:b24vk83) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L169)):*
> ```cpp
>   RotationMatrix<double> R3 = RotationMatrix<double>::MakeRotationMatrixX(q(2));
>   RotationMatrix<double> R_expected = R1 * R2 * R3;
>   RotationMatrix<double> R_bodyZYX =
> ```

I can see how we could have the same problem here as I point out above. 
What about comparing against `AngleAxisd(roll, Vector3d::UnitX()) * AngleAxisd(pitch, Vector3d::UnitY()) * AngleAxisd(yaw, Vector3d::UnitZ())`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-:-L3F0Q04UA11M6mtZhdd:b-uqunos)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Platform <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with some btws.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 116 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3FFsBO_yf504pfH2Yb:-L3FFsBO_yf504pfH2Yc:b-9dpexj) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L116)):*
> ```objc
>   /// ```
>   ///        ⎡cos(y) -sin(y)  0⎤   ⎡ cos(p)  0  sin(p)⎤   ⎡1      0        0 ⎤
>   /// R_AB = ⎢sin(y)  cos(a)  0⎥ * ⎢     0   1      0 ⎥ * ⎢0  cos(r)  -sin(r)⎥
> ```

BTW a -> y

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 118 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3FFytURxspKgsKYPpQ:-L3FFytURxspKgsKYPpR:bu1a12d) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L118)):*
> ```objc
>   /// R_AB = ⎢sin(y)  cos(a)  0⎥ * ⎢     0   1      0 ⎥ * ⎢0  cos(r)  -sin(r)⎥
>   ///        ⎣    0       0   1⎦   ⎣-sin(p)  0  cos(p)⎦   ⎣0  sin(r)   cos(r)⎦
>   /// ```
> ```

BTW this is beautiful documentation! But it makes me wonder why there isn't matching documentation for MakeRotationMatrixSpaceXYZ() ?

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 124 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzsxzNVih5vnAKu35:-L3FGhJciU0jCvYA9QwM:btgdkn2) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L124)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

this test could have the problem that you could've exactly copied the doc from `MakeRotationMatrixX()`.
I think a better test would compare against `Eigen::AngleAxisd(theta, Vector3d::UnitX()).matrix()`
</blockquote></details>

BTW good point. Probably have to compare with an epsilon though.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-:-L3FGwJ8JsiBKBqHaqYR:bfmioiq)*
<!-- Sent from Reviewable.io -->

mitiguy commented:
Review status: 1 of 3 files reviewed at latest revision, 18 unresolved discussions.

---

*[math/rotation_matrix.h, line 190 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EXaIf7PydATo6Rj0f:-L3FLIXahLrWDzcele2g:bviqzuv) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/math/rotation_matrix.h#L190)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

regarding the documentation of these methods. If they return the matrix for `R_AB`, does that mean that B is rotated `theta` radians about A's X axis? or viceversa? probably worth clarifying?
</blockquote></details>

Done.  Good suggestion.  Done for each of the simple rotations.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 55 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EY2uVaaxW8R87DOEU:-L3FLw-vezcIQCfRVtQm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L55)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

`R_AB` with single quotes?
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 60 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EYWFI3vu9d1OwVQpH:-L3FMmEXmg4MpBeYBG_Y:b-7rj6y0) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L60)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

I don't really know what theta*Bx is. Maybe say " ... of `theta` about `Bx`, where theta is defined positive in accordance to the right hand rule." (please don't forget single quotes around math symbols)
</blockquote></details>

Done.  Reworded.  Thanks for the suggestion.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 70 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EwKEQkvVoWEjtz0Fe:-L3FMsuSIWA1-6OQWZH7:bgvi6wd) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L70)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

ditto comments for `MakeRotationMatrixX()`
</blockquote></details>

Done.  Fixed similarly to MakeRotationMatrixX.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 78 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EvlN0qyJQdZ0sPBZw:-L3FNLqJM_6jqjfU-Nxu:b-kv0tfn) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L78)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

I would say: "See documentation for MakeRotationMatrixX() for definitions on the frames A and B related to this rotation and for `theta` sign convention". Otherwise I would just prefer to duplicate the discussion on A and B here to make all docs look alike since it really doesn't hurt, is not such a lengthy piece of doc anyways.
</blockquote></details>

Done.  I agree duplication is preferable.  It is better not having to bounce around.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 83 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EwMnBydjDAcJVwLNV:-L3FNSwwm6OGyMk-UUMP:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L83)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

ditto comments for `MakeRotationMatrixX()`
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 97 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Exdo_IFCinReMYSyI:-L3FO3GPErcuZbneU6Vm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L97)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

math between single quotes.
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 107 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey0gLVlpJUzm84MTh:-L3FO4BoPqfnqoAUrjFK:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L107)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

singles quotes for math
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 109 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey-8ppudthAZ3wnSO:-L3Ut4lJmmtYJmh0WoQc:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L109)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

singles quotes for math
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3Ey23-CPTxHn0aTf1Z:-L3Ut5mWzrde0hzlY9qk:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

> `R_AB`
singles quotes for math
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EyE1QefOZOoSFDaAF:-L3Uw4aJZLQTUa4to_Hn:bo875xj) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

 "sucessive right-handed rotations relative to A" 
This is confusing. Each rotation is relative to an intermediate frame, there are four frames here:
`R_AB = R_AB1 * R_B1B2 * R_B2B`.

</blockquote></details>

Done.  See new documentation.
I like your suggestion to introduce the 4 frames.  Please take a look.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EyvtBr6uE-jC3Ukxs:-L3UwTS3epPVKGBIk_y3:bn64tac) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

"y*Bz, then p*By, then r*Bx." I find this notation strange. I think it actually confuses rather than clarifying anything. It might be true for differential rotations (i.e. true also for angular velocities) but not for finite rotations.
</blockquote></details>

Done.  See previous comment.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 280 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzRRn5QCsHWrbUT9V:-L3Ux2_FvQTkmv9J_g_0:b-jh8jci) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L280)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

I'd say: "@returns `true` if each element of `this` exactly equals the elements in `other`".
Mathematically both are of course equal. The difference is when we perform floating point checks
</blockquote></details>

Done.  Good suggestion.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 282 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzJ1j-Fk38haa8TBd:-L3UxSIBN2WMeZzp0-Tn:b-naihrg) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/rotation_matrix.h#L282)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

I think you'd like to do ` return  matrix() == other.matrix()` here?
</blockquote></details>

Done.  Nice improvement.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 124 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3EzsxzNVih5vnAKu35:-L3UyYWY2CuNWUExnGeL:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L124)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW good point. Probably have to compare with an epsilon though.
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 141 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-RG7tV_CdHYDXYb_:-L3UyZdLEBjIvbhvi3AG:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L141)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

ditto. Test against `Eigen::AngleAxisd(theta, Vector3d::UnitY()).matrix()`?
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 156 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-UBeQnS1VJ6DDWYy:-L3Uy_VQiIh7UanPhLfp:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L156)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

ditto, test against `Eigen::AngleAxisd(theta, Vector3d::UnitZ()).matrix()`?
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/test/rotation_matrix_test.cc, line 169 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3F-_9jwxfxkrNyHixJ:-L3ZpoY6noi9gDmkK9Q9:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/94263d1d738ca2da5933d9ec9cfea61e386e6197/multibody/multibody_tree/math/test/rotation_matrix_test.cc#L169)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

I can see how we could have the same problem here as I point out above. 
What about comparing against `AngleAxisd(roll, Vector3d::UnitX()) * AngleAxisd(pitch, Vector3d::UnitY()) * AngleAxisd(yaw, Vector3d::UnitZ())`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 21 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 101 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_axQryAHYnp0HyW_c:-L3_axQryAHYnp0HyW_d:b8s4cwj) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L101)):*
> ```objc
>   /// Makes the %RotationMatrix for a Body-fixed (intrinsic) Z-Y-X rotation by
>   /// "yaw-pitch-roll" angles `[q(0), q(1), q(2)]`, which is equivalent to a
>   /// Space-fixed X-Y-Z rotation by roll-pitch-yaw angles `[q(2), q(1), q(0)]`.
> ```

BTW nit: you have `"yaw-pitch-roll"` in double quotes but `roll-pitch-yaw` is not.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_bYzfjnAGRBeALKKj:-L3_bYzfjnAGRBeALKKk:b-vuvong) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
> ```objc
>   /// R_AD = ⎢sin(y)  cos(y)  0⎥ * ⎢     0   1      0 ⎥ * ⎢0  cos(r)  -sin(r)⎥
>   ///        ⎣    0       0   1⎦   ⎣-sin(p)  0  cos(p)⎦   ⎣0  sin(r)   cos(r)⎦
>   ///      =       R_AB          *        R_BC          *        R_CD
> ```

The above is a roll-pitch-yaw sequence! The text below makes sense and describes a yaw-pitch-roll sequence but seems completely divorced from the rotation matrices shown above.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 138 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_cCGxkYg_EOXC2ley:-L3_cCGxkYg_EOXC2lez:bijhb0o) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L138)):*
> ```objc
>   /// R_AD = ⎢sin(y)  cos(y)  0⎥ * ⎢     0   1      0 ⎥ * ⎢0  cos(r)  -sin(r)⎥
>   ///        ⎣    0       0   1⎦   ⎣-sin(p)  0  cos(p)⎦   ⎣0  sin(r)   cos(r)⎦
>   ///      =       R_AB          *        R_BC          *        R_CD
> ```

BTW Perfect! This is a roll-pitch-yaw sequence and the rotation matrices perfectly match the text below.

I don't get how the two identical rotation matrix compositions can represent the two different orderings described in the text.

Does this make sense to you, @amcastro-tri ? Paul and I have discussed it a lot but haven't gotten anywhere.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
The documentation looks great @mitiguy! It is very difficult to convey these concepts in text. I added more details in our email discussion on how/why the same mathematical expression end up representing the two rotation conventions.

I am happy as is but I do see how I am biased now that you made me think about it and I finally get it (thanks for that!). There might still be room from improvement. 

---

Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 54 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3cg8tCE2PmOfpANWcp:-L3cg8tCE2PmOfpANWcq:btxdil0) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L54)):*
> ```objc
>   }
> 
>   /// Makes the %RotationMatrix `R_AB` associated with rotating frame B relative
> ```

BTW maybe "**a** frame B", since it's the first time it gets mentioned.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3cgHJvQFhSJmLm3MjB:-L3cgHJvQFhSJmLm3MjC:bdw412t) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L55)):*
> ```objc
> 
>   /// Makes the %RotationMatrix `R_AB` associated with rotating frame B relative
>   /// to frame A by an angle `theta` about unit vector `Ax = Bx` as shown below.
> ```

BTW maybe "**a** frame A", since it's the first time it gets mentioned.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 58 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3cgs9Eb_SN5shFYcG3:-L3cgs9Eb_SN5shFYcG4:b-tfwarf) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L58)):*
> ```objc
>   /// @param[in] theta radian measure of rotation angle about Ax.
>   /// @note `R_AB` relates two frames A and B having unit vectors Ax, Ay, Az and
>   /// Bx, By, Bz.  Initially, `Bx = Ax`, `By = Ay`, `Bz = Az`, then B undergoes
> ```

BTW, single quotes around (multiple characters) math symbols like `Bx` and `By`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 69 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3ch2NPZJCbHm4PiR57:-L3ch2NPZJCbHm4PiR58:b-sqmiz2) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L69)):*
> ```objc
>   }
> 
>   /// Makes the %RotationMatrix `R_AB` associated with rotating frame B relative
> ```

BTW. ditto formatting comments in `MakeRotationMatrixX()`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 84 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3chDbPV0YsNzfeadIV:-L3chDbPV0YsNzfeadIW:b-sqmiz2) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L84)):*
> ```objc
>   }
> 
>   /// Makes the %RotationMatrix `R_AB` associated with rotating frame B relative
> ```

BTW. ditto formatting comments in `MakeRotationMatrixX()`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_bYzfjnAGRBeALKKj:-L3clPjivgUuuwZwEU0w:b-vycte0) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

The above is a roll-pitch-yaw sequence! The text below makes sense and describes a yaw-pitch-roll sequence but seems completely divorced from the rotation matrices shown above.
</blockquote></details>

The description below seems consistent to me. It actually says the same thing you wrote in your email @sherm1 (with intermediary frames). no?

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3ck-Lf2bompM1H7285:-L3ck-Lf2bompM1H7286:b-egr9bs) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
> ```objc
>   /// ```
>   /// One way to visualize this rotation sequence is by introducing intermediate
>   /// frames B and C (A and D are real frames, B and C are not).
> ```

humm... to me B and C are as real as A and D. I'd just remove this parenthesized comment.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 122 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3ci4HFaif72nl3IiKC:-L3ci4HFaif72nl3IiKD:b-9ocdoq) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L122)):*
> ```objc
>   /// @li 3rd rotation R_CD: %Frame D rotates relative to frame C by a roll
>   /// angle `r` about `Cx = Dx`.
>   static RotationMatrix<T> MakeRotationMatrixBodyZYX(const Vector3<T>& q) {
> ```

BTW I like the notation in code actually. Based on that, why not to rename `q` to `yaw_pitch_roll`? or `ypr` if you want a shorter version.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 124 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3cjl7_gOm6zBrxYBwC:-L3cjl7_gOm6zBrxYBwD:b-7ea2v1) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L124)):*
> ```objc
>   static RotationMatrix<T> MakeRotationMatrixBodyZYX(const Vector3<T>& q) {
>     const Vector3<T> roll_pitch_yaw(q(2), q(1), q(0));
>     return RotationMatrix(math::rpy2rotmat(roll_pitch_yaw));
> ```

BTW, maybe even `return MakeRotationMatrixSpaceXYZ(roll_pitch_yaw)`

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 138 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_cCGxkYg_EOXC2ley:-L3d0U7jZAMD37YdRC6H:b-15tigp) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L138)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW Perfect! This is a roll-pitch-yaw sequence and the rotation matrices perfectly match the text below.

I don't get how the two identical rotation matrix compositions can represent the two different orderings described in the text.

Does this make sense to you, @amcastro-tri ? Paul and I have discussed it a lot but haven't gotten anywhere.
</blockquote></details>

Paul really nailed it here. Yes, the two identical matrix compositions do represent the same "concepts". I add a lot more of detail in the email discussion and I even propose how we could expand this to reflect that. VC is probably an option if you want, I can now very clearly see the how these things get composed together in a very graphical way for the two versions. Thank you for making me think about it!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-:-L3d1KqDf1Eorjr9k0Ie:b-pne8ie)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_bYzfjnAGRBeALKKj:-L3d9_WLBnVUpGcUfDMd:b5dfyvl) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

The description below seems consistent to me. It actually says the same thing you wrote in your email @sherm1 (with intermediary frames). no?
</blockquote></details>

I don't see how that's possible, @amcastro-tri, since my three rotation matrices contained different numbers and appeared in the opposite order. Take a look again at my note.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801)*
<!-- Sent from Reviewable.io -->

mitiguy commented:
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 110 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_bYzfjnAGRBeALKKj:-L4N17MY7MSqoB8SuHqD:b-akjrhb) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L110)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

I don't see how that's possible, @amcastro-tri, since my three rotation matrices contained different numbers and appeared in the opposite order. Take a look again at my note.
</blockquote></details>

Done.  Added a TODO to explain the sequence with the Sherm/Goldstein point-of-view.
What is written is one way to properly visualize the do the mathematics.
I think it is correct as written - but has a different starting perspective.
There are multiple ways to understand (and do the mathematics) for these sequences.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 113 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3ck-Lf2bompM1H7285:-L4N0_0mZ_A1cBjzG3ly:by56b8j) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L113)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

humm... to me B and C are as real as A and D. I'd just remove this parenthesized comment.
</blockquote></details>

Done.  Reworded

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 138 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L3_cCGxkYg_EOXC2ley:-L4N38a_E87FqFjBfw0X:b-akjrhb) ([raw file](https://github.com/robotlocomotion/drake/blob/81bf4058d5a9799a4390c86b66bdc0177c1673a8/multibody/multibody_tree/math/rotation_matrix.h#L138)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

Paul really nailed it here. Yes, the two identical matrix compositions do represent the same "concepts". I add a lot more of detail in the email discussion and I even propose how we could expand this to reflect that. VC is probably an option if you want, I can now very clearly see the how these things get composed together in a very graphical way for the two versions. Thank you for making me think about it!
</blockquote></details>

Done.  Added a TODO to explain the sequence with the Sherm/Goldstein point-of-view.
What is written is one way to properly visualize the do the mathematics.
I think it is correct as written - but has a different starting perspective.
There are multiple ways to understand (and do the mathematics) for these sequences.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Still <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>!

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-:-L4NdTVbQZ0oYp-Se4Rk:b-dbkjmw)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Excellent @mitiguy, thank you.
It'd be awesome if you could figure out a way to link @sherm1's view of rotations with your view. I was thinking a good way to do that would be going through the exercise of implementing and documenting `RotationMatrix::AngleAxis()`.
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 100 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRspgUhjBf0sqDct-:-L4aRspgUhjBf0sqDct0:b-jf5rw7) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L100)):*
> ```objc
> 
>   /// Makes the %RotationMatrix for a Body-fixed (intrinsic) Z-Y-X rotation by
>   /// "yaw-pitch-roll" angles `[q(0), q(1), q(2)]`, which is equivalent to a
> ```

no need for `q` anymore

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 104 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRTprmD4HHfSqZ-D6:-L4aRTprmD4HHfSqZ-D7:b-1h4ktj) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L104)):*
> ```objc
>   /// `[q(2), q(1), q(0)]`.
>   /// @param[in] ypr radian measures of three angles [yaw, pitch, roll].
>   /// @note Denoting yaw `y = q(0)`, pitch `p = q(1)`, roll `r = q(2)`, this
> ```

there is no more `q` in this doc (which I really like actually)

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 123 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRkqd-cOjKgKOyoyg:-L4aRkqd-cOjKgKOyoyh:b-equ6kd) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L123)):*
> ```objc
>   /// @li 3rd rotation R_CD: %Frame D rotates relative to frame C by a roll
>   /// angle `r` about `Cx = Dx`.
>   /// TODO(@mitiguy) Add Sherm/Goldstein's way to visualize rotation sequences.
> ```

BTW yes please! that'd be super useful!

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 130 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRvneFNXlh7vtQlG_:-L4aRvneFNXlh7vtQlGa:b-95tgt7) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L130)):*
> ```objc
> 
>   /// Makes the %RotationMatrix for a Space-fixed (extrinsic) X-Y-Z rotation by
>   /// "roll-pitch-yaw" angles `[q(0), q(1), q(2)]`, which is equivalent to a
> ```

BTW no more q

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 133 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRxTHlMb8ttgRYDS5:-L4aRxTHlMb8ttgRYDS6:b-95tgt7) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L133)):*
> ```objc
>   /// Body-fixed Z-Y-X rotation by "yaw-pitch-roll" angles `[q(2), q(1), q(0)]`.
>   /// @param[in] rpy radian measures of three angles [roll, pitch, yaw].
>   /// @note Denoting roll `r = q(0)`, pitch `p = q(1)`, yaw `y = q(2)`, this
> ```

BTW no more q

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-:-L4aSD3IrusyH2KQlkp7:bb14w0d)*
<!-- Sent from Reviewable.io -->

mitiguy commented:
Review status: 2 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 100 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRspgUhjBf0sqDct-:-L4bkobIltyrselVvQdD:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L100)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

no need for `q` anymore
</blockquote></details>

Done.

---

*[multibody/multibody_tree/math/rotation_matrix.h, line 104 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7801#-L4aRTprmD4HHfSqZ-D6:-L4bkpOud0SMWx8_2w_9:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d8b53989f5bfe91c7dcf2ddab6b66f327a736418/multibody/multibody_tree/math/rotation_matrix.h#L104)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

there is no more `q` in this doc (which I really like actually)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7801)*
<!-- Sent from Reviewable.io -->
